### PR TITLE
Feat/needs seo merge

### PR DESCRIPTION
**Pull request by jamiemonksuk**
This merges the (already reviewed) custom head tags branch, applies some small tweaks based on review, then builds on that by extending Needs a little, and pushing SocialMeta display behaviour over into Needs.

I've tested this with a local patch on the TheyStay repo and it seems to work nicely

### CODE REVIEW

**Comment by jamiemonksuk** [comment]
Just a note, you might want to review this commit by commit after the initial merge, which remains unchanged from the previous review at that point

File: src/sprout/Controllers/Admin/PageAdminController.php [hunk 1]
```diff
@@ -403,7 +404,7 @@ public function _addSave(&$page_id)
         Pdb::insert('page_revisions', $update_fields);
 
         // Admin permissions
-        if ($_POST['admin_perm_specific'] == 1 and @count($_POST['admin_permissions'])) {
```
Comment (gwillz) [inline src/sprout/Controllers/Admin/PageAdminController.php:406]: :scream: I thought we already cleaned these out!

**Review by gwillz — CHANGES_REQUESTED** [review]
The homepage methods are completely broken without having the `homepage_custom_tags` table in the schema.

Perhaps that module simply needs it's own helper (that extends this one) for managing it's tags.

Understandably we pretty much always have a homepage module, but core needs to function on it's own without these loose dependencies.

File: src/sprout/Controllers/Admin/PageAdminController.php (hunk 1 above)
Comment (jamiemonksuk) [reply src/sprout/Controllers/Admin/PageAdminController.php:406]: Yeah that was a surprise!

**Comment by jamiemonksuk** [comment]
> `homepage_custom_tags`

Maybe for now we pull them out, then look at it as a new piece?

**Comment by jamiemonksuk** [comment]
I split them out into a new helper as proposed, seems cleaner too

**Review by gwillz — APPROVED** [review]
Feeling good!

**Comment by jamiemonksuk** [comment]
@gwillz I rejigged it to be table agnostic as per @aitken85 suggestion :)

**Pull request merged**